.github/workflows/sanitisers.yml

```diff
@@ -0,0 +1,225 @@
+name: Sanitiser Checks
+
+# on: workflow_dispatch
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+    types: [opened, synchronize, reopened, ready_for_review]
+
+jobs:
+  conan-cache:
+    if: github.event.pull_request.draft == false
+    runs-on: ubuntu-latest
+    env:
+      HOST_TYPE: ci
+      REDIS_QUEUE_HOST: redis
+      REDIS_STATE_HOST: redis
+    container:
+      image: faasm/faabric:0.2.2
+    defaults:
+      run:
+        working-directory: /code/faabric
+    services:
+      redis:
+        image: redis
+    steps:
+      # --- Code update ---
+      - name: "Fetch ref"
+        run: git fetch origin ${GITHUB_REF}:ci-branch
+      - name: "Check out branch"
+        run: git checkout --force ci-branch
+      # --- Set-up ---
+      - name: "Ping redis"
+        run: redis-cli -h redis ping
+      # --- Cache based on Conan version and ExternalProjects cmake source
+      - name: Get Conan version
+        id: get-conan-version
+        run: |
+          echo "::set-output name=conan-version::$(conan --version | tr -d '[:alpha:] ')"
+        shell: bash
+      - uses: actions/cache@v2
+        with:
+          path: '~/.conan'
+          key: ${{ runner.os }}-${{ steps.get-conan-version.outputs.conan-version }}-${{ hashFiles('cmake/ExternalProjects.cmake') }}
```

Comment:
430 MB, the limit is 10GB per repo (<https://github.com/actions/cache>), and this should be a cache hit most times as the dependency versions don't change all that often

Reply:
I also think this will be a great speedup to the runtime of general tests if we use the caching there (#187), but @kubasz it looks as though we are using the wrong key: https://github.com/faasm/faabric/runs/4337385997?check_suite_focus=true#step:7:4

(if the logs are to be trusted of course)

could you double-check?

Reply:
Huh, seems that hashFiles doesn't work, let me look into it

Reply:
I had to replace it with manual hashing in bash+sha256sums, it works in the most recent commit. @csegarragonz This change will probably need porting over to #187